packaging/docker/Dockerfile

```diff
@@ -13,35 +21,42 @@ RUN mkdir crates
 RUN cargo new --lib crates/pedersen
 # Correct: --lib. We'll handle the binary later.
 RUN cargo new --lib crates/pathfinder
-COPY Cargo.toml Cargo.toml
-
-COPY crates/pathfinder/Cargo.toml crates/pathfinder/Cargo.toml
-COPY crates/pedersen/Cargo.toml crates/pedersen/Cargo.toml
-COPY crates/pedersen/benches crates/pedersen/benches
+RUN cp /tmp/pathfinder/Cargo.toml Cargo.toml
 
-RUN RUSTFLAGS='-L/usr/lib -Ctarget-feature=-crt-static' cargo build --release --target=x86_64-unknown-linux-musl
+RUN cp /tmp/pathfinder/crates/pathfinder/Cargo.toml crates/pathfinder/Cargo.toml
+RUN cp /tmp/pathfinder/crates/pedersen/Cargo.toml crates/pedersen/Cargo.toml
+RUN cp -R /tmp/pathfinder/crates/pedersen/benches crates/pedersen/benches
 
+RUN RUSTFLAGS='-L/usr/lib -Ctarget-feature=-crt-static' cargo build --release
 
 # Compile the actual libraries and binary now
-COPY . .
+RUN cp -R /tmp/pathfinder/* .
 
-# Mark these for re-compilation 
+# Mark these for re-compilation
 RUN touch crates/pathfinder/src/lib.rs
 RUN touch crates/pedersen/src/lib.rs
 
-RUN RUSTFLAGS='-L/usr/lib -Ctarget-feature=-crt-static' cargo build --release --target=x86_64-unknown-linux-musl --bin pathfinder
+RUN RUSTFLAGS='-L/usr/lib -Ctarget-feature=-crt-static' cargo build --release
+
+# Copy binary to hardcoded path so we can copy it to the `runner` image later on
+RUN mkdir bins && cp $(find ./target -name 'pathfinder') ./bins/pathfinder
 
-#######################################
 # Stage 2: Build the Python libraries #
 #######################################
 FROM python:3.8-alpine AS python-builder
 
 RUN apk add --no-cache gcc musl-dev gmp-dev g++
 
+# bring in the pathfinder repo from the previous build
+COPY --from=rust-builder /tmp/pathfinder /tmp/pathfinder
```

Comment:
The advantage to building the Python libs in its own stage is that when building with `nerdctl` or even `docker buildx`, it will parallelize stage 1 and 2. If we need to wait until `rust-builder` is done, then it breaks this parallelization.

Reply:
Not sure what you mean by this. I did not make any changes to this stage.

Reply:
The addition of the `COPY` [line 51](https://github.com/eqlabs/pathfinder/pull/200/files#diff-028a7f6bae25b75ad5b6f936e31c3d30bf50ce15af5cc2be41cf23af5cdd7805R51) will require Stage 2 to wait until Stage 1 is complete, rather than Stage 1 + Stage 2 building in parallel. Building Rust in alpine is already slow enough and this would add a lot more time.

Also [these lines](https://github.com/eqlabs/pathfinder/pull/200/files#diff-028a7f6bae25b75ad5b6f936e31c3d30bf50ce15af5cc2be41cf23af5cdd7805R55-R59) were changed